child-project/src/test/java/com/example/SimpleTest.java

```diff
@@ -8,11 +8,13 @@
 
 @RunWith(JUnit4.class)
 public class SimpleTest {
+  // With "--initialize-at-build-time=com.example", this initialization runs at build time.
+  static final String myProp= System.getProperty("my.prop");
```

Comment:
This is where "code that is part of your application but executed at image build time" means in https://www.graalvm.org/22.0/reference-manual/native-image/Properties/